File: examples/typegraphs/jwt.py
# Copyright Metatype OÜ, licensed under the Mozilla Public License Version 2.0.
# SPDX-License-Identifier: MPL-2.0

# skip:start
from typegraph import Graph, Policy, t, typegraph
from typegraph.graph.params import Auth, Cors
from typegraph.runtimes.deno import DenoRuntime

# skip:end


@typegraph(
    # skip:start
    cors=Cors(allow_origin=["https://metatype.dev", "http://localhost:3000"]),
    # skip:end
)
def jwt_authentication(g: Graph):
    deno = DenoRuntime()
    public = Policy.public()

    ctx = t.struct({"your_own_content": t.string().optional()})
    # highlight-next-line
    g.auth(Auth.hmac256("custom"))

    g.expose(
        get_context=deno.identity(ctx).apply(
            {
                "your_own_content": g.from_context("your_own_content"),
            },
        ),
        default_policy=[public],
    )

File: examples/typegraphs/jwt.ts
// skip:start
import { Policy, t, typegraph } from "@typegraph/sdk/index.ts";
import { Auth } from "@typegraph/sdk/params.ts";
import { DenoRuntime } from "@typegraph/sdk/runtimes/deno.ts";

// skip:end

typegraph(
  {
    name: "jwt-authentication",
    // skip:next-line
    cors: { allowOrigin: ["https://metatype.dev", "http://localhost:3000"] },
  },
  (g) => {
    const deno = new DenoRuntime();
    const pub = Policy.public();

    const ctx = t.struct({
      your_own_content: t.string().optional(),
    });
    // highlight-next-line
    g.auth(Auth.hmac256("custom"));

    g.expose(
      {
        get_context: deno.identity(ctx).apply({
          your_own_content: g.fromContext("your_own_content"),
        }),
      },
      pub,
    );
  },
);
